low-level/framework/utils/dmidecode_interface.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2019-2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+Module which provides system information using 'dmidecode' command
+"""
+
+import re
+import psutil
+
+from pathlib import Path
+from statistics import mean
+from framework.utils.utility import Utility
+from framework.utils.service_logging import logger
+from cortx.utils.process import SimpleProcess
+
+class Dmidecode(Utility):
+    """
+    Interface class to retrieve system's hardware related information
+    using 'dmidecode' command.
+    """
+    DMIDECODE = "sudo /sbin/dmidecode"
+
+    def __init__(self):
+        """init method"""
+        super(Dmidecode, self).__init__()
+
+    def get_cpu_info(self):
+        """Returns online cpus list"""
+        try:
+            cmd = self.DMIDECODE + " -t 4"
+            response, err, _ = SimpleProcess(cmd).run()
+            if err:
+                logger.error("Failed to get list of Online CPUs."
+                             f"ERROR:{err}")
+                return
+            matches = re.findall("Socket Designation:.*|"
+                                 "Status:.*", response.decode())
+
+            cpu_map = {}
+            cpu_list = []
+            cpu_present = cpu_status = None
+            while matches:
+                item = matches.pop(0)
+                if "Designation:" in item:
+                    cpu_str = item.strip().split(": ")[1]
+                    cpu_dig = re.findall('\d+', cpu_str )
```

Comment:
`r('\d')` also remove the blank line before `)`

Reply:
I understand that. But make it as a raw string by adding `r` before the pattern. Because sometimes python will try to interpret `\d` as escape sequence. We just want to make sure that only re module parses it. Also pls remove the space before `)`